src/ui/public/notify/__tests__/notifier.js

```diff
@@ -331,6 +332,155 @@ describe('Notifier', function () {
+      }
+      expect(callDirectiveIncorrectly).to.throwException(function (e) {
+        expect(e.message).to.be('Directive param is required, and must be an object');
+      });
+    });
+
+    it('throws if second param is not an object', function () {
+      // destroy the default custom notification, avoid duplicate handling
+      directiveNotification.clear();
+
+      function callDirectiveIncorrectly() {
+        const badConfigParam = null;
+        directiveNotification = notifier.directive(directiveParam, badConfigParam);
+      }
+      expect(callDirectiveIncorrectly).to.throwException(function (e) {
+        expect(e.message).to.be('Config param is required, and must be an object');
+      });
+    });
+
+    it('throws if directive param has scope definition instead of allow the helper to do its work', function () {
+      // destroy the default custom notification, avoid duplicate handling
+      directiveNotification.clear();
+
+      function callDirectiveIncorrectly() {
+        const badDirectiveParam = {
+          scope: {
+            garbage: '='
+          }
+        };
+        directiveNotification = notifier.directive(badDirectiveParam, customParams);
+      }
+      expect(callDirectiveIncorrectly).to.throwException(function (e) {
+        expect(e.message).to.be('Directive should not have a scope definition. Notifier has an internal implementation.');
+      });
+    });
+
+    it('throws if directive param has link function instead of allow the helper to do its work', function () {
+      // destroy the default custom notification, avoid duplicate handling
+      directiveNotification.clear();
+
+      function callDirectiveIncorrectly() {
+        const badDirectiveParam = {
+          link: ($scope) => {
+            /*eslint-disable no-console*/
+            console.log($scope.nothing);
+            /*eslint-enable*/
+          }
+        };
+        directiveNotification = notifier.directive(badDirectiveParam, customParams);
+      }
+      expect(callDirectiveIncorrectly).to.throwException(function (e) {
+        expect(e.message).to.be('Directive should not have a link function. Notifier has an internal link function helper.');
+      });
+    });
+
+    it('has a directive function to make notifications with template and scope', function () {
+      expect(notifier.directive).to.be.a('function');
+    });
+
+    it('sets the scope property and link function', function () {
+      expect(directiveNotification).to.have.property('directive');
+      expect(directiveNotification.directive).to.be.an('object');
+
+      expect(directiveNotification.directive).to.have.property('scope');
+      expect(directiveNotification.directive.scope).to.be.an('object');
+
+      expect(directiveNotification.directive).to.have.property('link');
+      expect(directiveNotification.directive.link).to.be.an('function');
+    });
+
+    // TODO
+    it.skip('renders the directive template', function () {
```

Comment:
@spalger I had to `.skip` this because this implementation doesn't work. Error is:

`Error: Injector already created, can not register a module!`

Note there's already a `ngMock.module` and `ngMock.inject`  in this test in the main `beforeEach`.

I'd like to be able to test that `render_directive` is internally doing the right thing with this notification type. Any suggestions on how to implement a test like that?